### PR TITLE
Create Gradle Orb

### PR DESCRIPTION
**Pull request by marcomorain**

### CODE REVIEW

File: src/gradle/orb.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+commands:
+  with_cache:
+    parameters:
+      steps:
+        type: steps
+    steps:
+      - run:
+          name: Generate Cache Checksum
+          commad: find . -name 'build.gradle' -exec cat {} + | shasum | awk '{print $1}' > /tmp/gradle_cache_seed
```
Comment (ndintenfass) [inline src/gradle/orb.yml]: typo: "commad"

File: src/gradle/orb.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  gradle:
+    description: The docker container to use when running Gradle builds
+    docker:
+      - image: 8-jdk-node
+    environment:
+      # Disable fancy terminal updates
+      TERM: dumb
+
+commands:
+  with_cache:
```
Comment (ndintenfass) [inline src/gradle/orb.yml]: Consider calling this `with_gradle_cache` given how specific it is to build.gradle

File: src/gradle/orb.yml (hunk 1 above)
Comment (ndintenfass) [reply src/gradle/orb.yml]: Also, I suggest putting in a comment to explain this line and why you aren't doing a checksum on build.gradle

File: src/gradle/orb.yml (hunk 2 above)
Comment (marcomorain) [reply src/gradle/orb.yml]: The orb is already called `gradle`, so the user will be typing `gradle/with_cache`:

File: src/gradle/orb.yml (hunk 1 above)
Comment (marcomorain) [reply src/gradle/orb.yml]: 👍

File: src/gradle/orb.yml (hunk 2 above)
Comment (zzak) [reply src/gradle/orb.yml]: +1 to marc's naming

**Comment by marcomorain** [comment]
Merging to get this over the line. We can iterate on the orb.

**Pull request merged**